src/stores/ui.ts

```diff
@@ -26,8 +71,62 @@ export const useUiStore = defineStore('ui', {
         this.instructionsSnackbarState = 'visible'
       }
     },
+
     hideInstructionsSnackbar() {
       this.instructionsSnackbarState = 'dismissed'
     },
+    /**
+     * Given a list of key value pairs of UI state parameters and their states,
+     * populate the store state to match the cookie.
+     *
+     * @param cookies - mapping of UI state parameters and their states.
+     */
+    initFromCookies(cookies: OpenverseCookies) {
+      this.isDesktopLayout = cookies.uiIsDesktopLayout ?? false
+      this.isFilterDismissed = cookies.uiIsFilterDismissed ?? false
+
+      // Middleware sets the cookie value before calling `initFromCookies`.
+      this.isMobileUa = cookies.uiIsMobileUa ?? false
+      this.innerFilterVisible = this.isDesktopLayout
+        ? !this.isFilterDismissed
+        : false
+    },
+
+    /**
+     * If the breakpoint is different from the state, updates the state, and saves it into app cookies.
+     *
+     * @param isDesktopLayout - whether the layout is desktop (`lg` with the `new_header`
+     * and `md` with the `old_header`).
+     * @param setCookieFn - sets the app cookie.
+     */
+    updateBreakpoint(isDesktopLayout: boolean, setCookieFn: CookieSetter) {
```

Comment:
We could add a check for `isMobileUa` here and only set the `isDesktopLayout` to true if `isMobileUa` is false. However, I'm not sure if it makes sense to use a mobile layout for `lg` mobile screens.

Reply:
I think you're right, that it doesn't make sense. To me, it would make sense to rely exclusively on the breakpoint information _if we have it_ and fall back to UA sniffing only when we don't yet know any breakpoint information from the cookie.